cleanlab/regression/rank.py

```diff
@@ -0,0 +1,153 @@
+        "TO_BE_NAMED": get_score_to_named_for_each_label,  # TODO - update name once finalised
+    }
+
+    # TODO - update name once finalised
+    try:
+        scoring_func = scoring_funcs[method]
+    except KeyError:
+        raise ValueError(
+            f"""
+            {method} is not a valid scoring method.
+            Please choose a valid scoring technique: {scoring_funcs.keys()}.
+            """
+        )
+
+    # Calculate scores
+    label_quality_scores = scoring_func(labels, predictions)
+    return label_quality_scores
+
+
+def get_residual_score_for_each_label(
+    labels: np.ndarray,
+    predictions: np.ndarray,
+) -> np.ndarray:
+    """Returns the residual based label-quality scores for each datapoints.
+
+    This is function to compute label-quality scores for regression datasets,
+    where lower score indicate labels less likely to be correct.
+
+    Residual based scores can work better for datasets where independent variables
+    are based out of normal distribution.
+
+    Parameters
+    ----------
+    labels: np.ndarray
+        Labels in the same format expected by the :py:func:`get_label_quality_scores <cleanlab.regression.rank.get_label_quality_scores>` function.
+
+    predictions: np.ndarray
+        Predicted labels in the same format expected by the :py:func:`get_label_quality_scores <cleanlab.regression.rank.get_label_quality_scores>` function.
+
+    Returns
+    -------
+    label_quality_scores: np.ndarray
+        Contains one score (between 0 and 1) per example.
+        Lower scores indicate more likely mislabled examples.
+
+    """
+    residual = predictions - labels
+    label_quality_scores = np.exp(-abs(residual))
+    return label_quality_scores
+
+
+# TODO - change name of the function
+# TODO - change name of function in test
+def get_score_to_named_for_each_label(
+    labels: np.ndarray,
+    predictions: np.ndarray,
+    *,
+    variance: float = 10,
+) -> np.ndarray:
+    """Returns label-quality scores.
+
+    This is function to compute label-quality scores for regression datasets,
+    where lower score indicate labels less likely to be correct.
+
+    Parameters
+    ----------
+    labels: np.ndarray
+        Labels in the same format expected by the :py:func:`get_label_quality_scores <cleanlab.regression.rank.get_label_quality_scores>` function.
+
+    predictions: np.ndarray
+        Predicted labels in the same format expected by the :py:func:`get_label_quality_scores <cleanlab.regression.rank.get_label_quality_scores>` function.
+
+    variance: float, default = 10
+        Manipulates variance of the distribution of residual.
+
+    Returns
+    -------
+    label_quality_scores: np.ndarray
+        Contains one score (between 0 and 1) per example.
+        Lower scores indicate more likely mislabled examples.
+    """
+
+    neighbors = int(np.ceil(0.1 * labels.shape[0]))
+    knn = NearestNeighbors(n_neighbors=neighbors, metric="euclidean")
```

Comment:
The construction of the `knn` object should be closer to the place it is first used.
The same applies to `neighbors`.

Reply:
addressed here: [f2c5862](https://github.com/cleanlab/cleanlab/pull/572/commits/f2c5862864c76f72113a244e7dc2320c1b574d28)